### ISSUE TITLE
Pyright doesn't know that instances of the metaclass have `__name__`

### ISSUE THREAD

**Issue by jack1142** [microsoft/pyright#2166]
**Describe the bug**
Pyright doesn't know that instances of the metaclass have `__name__` attribute (and perhaps others?).
It does appear to [be type-hinted properly in typeshed](https://github.com/python/typeshed/blob/master/stdlib/builtins.pyi#L138) and mypy doesn't generate any errors for the code below (and it does check the attribute names - making a typo will show an error)

**To Reproduce**
1. Run pyright on code below and see an error:
```
error: Cannot access member "__name__" for type "_Meta"
  Member "__name__" is unknown (reportGeneralTypeIssues)
```

**Expected behavior**
No errors.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class _Meta(type):
    def some_method(self) -> str:
        return self.__name__
```

**VS Code extension or command-line**
Running pyright 1.1.160 as a command-line tool. Also tried it out on `main` and it doesn't work there as well.

**Additional context**
N/A

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.161, which I just published. It will also be included in the next release of pylance.
